src/components/HomeScreen.js

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react';
+import {
+  TextInput,
+  TouchableOpacity,
+  Button,
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { connect } from 'react-redux';
+import {verifyUser} from '../store/UsersActions';
+
+class HomeScreen extends Component {
+  constructor(){
+    super();
+    this.state = {
+      email: "",
+      password: "",
+    }
+    this.onLogin = this.onLogin.bind(this);
+  }
+
+  onLogin (navigation){
+    const expression = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+    if(this.state.password.length > 8 && expression.test(this.state.email.toLowerCase())){
+      this.props.onLogin(this.state.email);
+      navigation.navigate('Details');
+    }
+    else{
+      alert("Invalid email or password");
+    }
+  } 
```

Comment:
Aqui não precisas de receber o naviagtion como param, podes fazer outra vez o destruct do props porque estas dentro da class.

O destruct é uma função muito util porque evita que uses o **this.state...** cada vez que queres aceder ao state do componente.

```suggestion
  onLogin() {
    const { password, email } = this.state
    const { navigation, onLogin } = this.props;
    const expression = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
    
    if (password.length > 8 && expression.test(email.toLowerCase())) {
      this.props.onLogin(this.state.email);
      navigation.navigate('Details');
    }
    else {
      alert("Invalid email or password");
    }
  } 
```